dll/win32/shell32/CShellDispatch.cpp

```diff
@@ -119,58 +128,83 @@ HRESULT STDMETHODCALLTYPE CShellDispatch::BrowseForFolder(LONG Hwnd, BSTR Title,
 HRESULT STDMETHODCALLTYPE CShellDispatch::Windows(IDispatch **ppid)
 {
     TRACE("(%p, %p)\n", this, ppid);
+    return CoCreateInstance(CLSID_ShellWindows, NULL, CLSCTX_LOCAL_SERVER, IID_PPV_ARG(IDispatch, ppid));
+}
 
-    *ppid = NULL;
+static HRESULT SHELL_OpenFolder(LPCITEMIDLIST pidl, LPCWSTR Verb = NULL)
+{
+    // Custom function because SHOpenFolderAndSelectItems does not support non-FS pidls nor verbs.
+    SHELLEXECUTEINFOW sei;
+    sei.cbSize = sizeof(sei);
+    sei.fMask = SEE_MASK_IDLIST | SEE_MASK_FLAG_DDEWAIT;
+    sei.hwnd = NULL;
+    sei.lpVerb = Verb;
+    sei.lpFile = sei.lpParameters = sei.lpDirectory = NULL;
+    sei.nShow = SW_SHOW;
+    sei.lpIDList = const_cast<LPITEMIDLIST>(pidl);
+    if (ShellExecuteExW(&sei))
+        return S_OK;
+    DWORD error = GetLastError();
+    return HRESULT_FROM_WIN32(error);
+}
 
-    return E_NOTIMPL;
+static HRESULT OpenFolder(VARIANT vDir, LPCWSTR Verb = NULL)
+{
+    CComHeapPtr<ITEMIDLIST> idlist;
+    HRESULT hr = VariantToIdlist(&vDir, &idlist);
+    if (hr == S_OK && SHELL_OpenFolder(idlist, Verb) == S_OK)
+    {
+        return S_OK;
+    }
+    return S_FALSE;
```

Comment:
Maybe just return an `hr` instead, to report the actual returned status back to the caller?
```suggestion
    return hr;
```

Reply:
No, Windows does not report an error if you give it a bad folder path here.